scipy/stats/_continuous_distns.py

```diff
@@ -5862,22 +5862,40 @@ def _pdf(self, x, df, nc):
         # ncx2.pdf(x, df, nc) = exp(-(nc+x)/2) * 1/2 * (x/nc)**((df-2)/4)
         #                       * I[(df-2)/2](sqrt(nc*x))
         cond = np.ones_like(x, dtype=bool) & (nc != 0)
-        return _lazywhere(cond, (x, df, nc), f=_ncx2_pdf, f2=chi2.pdf)
+        with warnings.catch_warnings():
+            message = "overflow encountered in _ncx2_pdf"
+            warnings.filterwarnings("ignore", message=message)
+            return _lazywhere(cond, (x, df, nc),
+                              f=_boost._ncx2_pdf, f2=chi2.pdf)
 
     def _cdf(self, x, df, nc):
         cond = np.ones_like(x, dtype=bool) & (nc != 0)
-        return _lazywhere(cond, (x, df, nc), f=_ncx2_cdf, f2=chi2.cdf)
+        return _lazywhere(cond, (x, df, nc), f=_boost._ncx2_cdf, f2=chi2.cdf)
 
     def _ppf(self, q, df, nc):
         cond = np.ones_like(q, dtype=bool) & (nc != 0)
-        return _lazywhere(cond, (q, df, nc), f=sc.chndtrix, f2=chi2.ppf)
+        with warnings.catch_warnings():
+            message = "overflow encountered in _ncx2_ppf"
+            warnings.filterwarnings("ignore", message=message)
+            return _lazywhere(cond, (q, df, nc),
+                              f=_boost._ncx2_ppf, f2=chi2.ppf)
+
+    def _sf(self, x, df, nc):
+        return _boost._ncx2_sf(x, df, nc)
```

Comment:
Does it work well when `nc=0`? If so, you can get rid of the `lazywhere`. If not, we should use `_lazywhere` here, too (so that the regular `chi2` distribution is used whenever the noncentrality parameter is zero). Same with `isf` below.

Reply:
Boost behaves badly when `nc=0`, so `lazywhere` should be used

Reply:
Huh, it didn't in my testing, but I added it anyway.